src/nsrt_learning.py

```diff
@@ -75,13 +77,25 @@ def segment_trajectory(trajectory: GroundAtomTrajectory) -> List[Segment]:
     for t in range(len(actions)):
         current_segment_traj[0].append(states[t])
         current_segment_traj[1].append(actions[t])
-        if all_atoms[t] != all_atoms[t+1]:
+        switch = all_atoms[t] != all_atoms[t+1]
+        if actions[t].has_option() and t < len(actions) - 1:
```

Comment:
I don't fully understand the second condition here, but tell me if this is right:
We don't care about the last option in the traj because there's no next option after it, so we could never possibly want to segment due to that option? But we still need to check atoms here, because the length of atoms is one longer.

Reply:
yes that's right, i'll add that as a comment thanks